test/core.jl

```diff
@@ -5089,3 +5089,23 @@ m22929_2.x = m22929_1
 @test !isdefined_22929_x(m22929_1)
 @test isdefined_22929_1(m22929_2)
 @test isdefined_22929_x(m22929_2)
+
+# issue 18933
+module GlobalDef18933
+    using Base.Test
+    f() = (global sin; nothing)
```

Comment:
It's a minor breaking change (but I suspect it will break a handful of packages anyways), but this PR means that:
```
let
   global function f(); return 1; end
end
```
Now expands to
```
global f
f() = 1
```
And thus will _not_ add to an existing binding from `import`, but will instead give an overwrite notice.